### ISSUE TITLE
Images get aren't shared (check_existing doesn't seem to work)

### ISSUE THREAD

**Issue by swift502** [Zarig/DragonFF#14]
Hi,

I'd really like to solve this because it's the last remaining piece for me.

The check_existing option used here doesn't seem to work:

https://github.com/Zarig/DragonFF/blob/40762f456a6b446e941ace4f7d32cdf1bed2415f/gta/dff_importer.py#L269-L273

![](https://user-images.githubusercontent.com/24359130/59460748-1ab9e980-8e20-11e9-8c14-401827364507.png)

I tried to find what exactly happens in Blender's source, but I ended up here, and don't really know where it leads. I can't find this "load" function anywhere:

https://github.com/dfelinto/blender/blob/78f8679cfc6d6e8f19d84908ab4343ce6638faf3/release/scripts/modules/bpy_extras/image_utils.py#L106

I tried all sorts of things and additional options and nothing seems to work.
Any ideas on what's causing this?

**Comment by swift502** [comment]
Ok I think the actual function is here.
https://github.com/dfelinto/blender/blob/483ae407d9d1b2b173face0efb96f2d287d51ea4/source/blender/blenkernel/intern/image.c#L566-L604

**Comment by swift502** [comment]
Ok I just noticed that even the materials get duplicated. I'll try manually checking for existing images and materials.

**Comment by Zarig** [comment]
Yeah, not sure what's wrong there. It seems to be checking for the images. Might depend on context, not sure about that either.

**Comment by swift502** [comment]
I hopefully solved it in a dirty (again) way.
https://github.com/swift502/DragonFF/commit/ed9c4042df955eb0458c068b977653590e69ed45

And to be honest, I don't even understand how the material merging works in your master, because the list which holds the loaded materials is reset on every dff import isn't it?

But I never noticed any duplicate materials when importing the maps, so I don't know. Maybe there just isn't that many shared materials.
